internal/cli/deployments/connect.go

```diff
@@ -287,12 +289,27 @@ func (opts *ConnectOpts) connectToAtlas() error {
 		return err
 	}
 
+	// Connectionstrings are empty when the server is not in IDLE
+	// r.ConnectionStrings.PrivateSrv == nil and r.ConnectionStrings.StandardSrv == nil
+	if r.StateName == nil || *r.StateName != "IDLE" {
```

Comment:
```suggestion
	if r.GetStateName() != "IDLE" {
```
https://github.com/mongodb/atlas-sdk-go/blob/main/docs/doc_5_best_practices.md#using-getters-instead-of-direct-field-access

Reply:
Thanks, I wasn't aware of these getters and setters.

[q] If we recommend using these getters and setters. Why do we expose the underlying fields to the end user?

Reply:
because go, you can't serialize/deserialize private fields

Reply:
but we can with `MarshalJSON` / `UnmarshalJSON`, I think we never paid attention to struct best practices in go `Get...` is considered against go's naming conventions https://go.dev/doc/effective_go#Getters